src/pinns_pde_solve.jl

```diff
@@ -782,9 +819,14 @@ function DiffEqBase.discretize(pde_system::PDESystem, discretization::PhysicsInf
     end
 
     function loss_function_(θ,p)
-        return pde_loss_function(θ) + bc_loss_function(θ)
+        if param_estim == false
```

Comment:
I can see a lot of cases where the additional_loss function is unrelated to parameter estimation, we should allow the user to pass it in and use it optimization independent of param_estim here.  The if statement should be whether or not `additional_loss isa Nothing`

Reply:
I think that the additional loss function should be excluded from the interface, for this there is a Low-level API.

Reply:
`additional_loss` is better renamed as `optim_p__loss` or something like that

Reply:
>I think that the additional loss function should be excluded from the interface, for this, there is a Low-level API.

I don't get what you mean by that. The low level API is for inspection. This is for injecting code into the loss.

>additional_loss is better renamed as optim_p__loss or something like that

There are many reasons for adding loss terms other than for optimization of parameters, like regularization.

Reply:
for this task, it seems to be better not to use an additional loss function, but to use a loss function specialized for this task to optimize the parameters. And we can not take it out to the external interface, but generate it under the hood. Add method `get_optimize_p_loss_function`.

Reply:
Actually, we have already encountered this https://github.com/SciML/NeuralPDE.jl/pull/136

Reply:
How do we pass data in that case? By defining a condition over each point in data?

Reply:
@KirillZubov we'll have to define a regularisation loss separately then?

Reply:
> How do we pass data in that case? By defining a condition over each point in data?

data can be passed to `PhysicsInformedNN` instead of loss func

Reply:
we can even create a separate structure that will store everything that needs to optimize the parameters - data and all the thing

Reply:
> > How do we pass data in that case? By defining a condition over each point in data?
> 
> data can be passed to `PhysicsInformedNN` instead of loss func

But then what will our eventual loss function be? It can be L2, L1, or anything that the user might want right?

Reply:
PhysicsInformedNN([chain1 , chain2, chain3], GridTraining(dt), EstimatiomParametersProblem(data, ...))

Reply:
> @KirillZubov we'll have to define a regularisation loss separately then?

what do you mean by a regularisation loss?

Reply:
If we want to add a regularisation term with the loss, such as Ridge Regression. Even ` EstimatiomParametersProblem(data, ...)` after we define this, we'll still need input for an optional/additional loss

Reply:
There isn't even 1 loss function for loss against data. There isn't 1 regularization method either. We tried this in DiffEqParamEstim.jl: you will not cover all loss functions. What about if they only have data for 2 of the 3 series? What if they have data only for the derivatives? It's completely impossible to handle like that. We already learned this, so I'm going to say that's a hard no. The user needs a hook to define a loss function which encloses data, there isn't a real alternative there.

Reply:
ok

Reply:
it would be nice to avoid explicit indexing inside the loss function and, in general, to make the interface as clear and compact as possible for the user.
```julia
function get_separete(chains)
    initθs = DiffEqFlux.initial_params.(chains)
    acum =  [0;accumulate(+, length.(initθs))]
    separete = [acum[i]+1 : acum[i+1] for i in 1:length(acum)-1]
end

sep = get_separete([chain1,chain2,chain3])

function additional_loss(phi, θ , p)
    _loss(u, t) = sum(abs2, phi[i](t , θ[s])[1] - (u[i])  for (i,s) in enumerate(sep))
    return sum(abs2, _loss(u,[t]) for (u , t) in data)/length(data)
end
```

Reply:
as @zoemcc   already mentioned, if this is an additional general purpose function then it should not depend on `param_estim` if not and this is a function only for parameter optimization then it should be renamed.

Reply:
it turns out, that `eqs` and `bcs` are in symbolic representation and there is no additional function. And we get different:  symbolic in one` x(t),  Dt(x(t)) β`, in other in inner  representation  `phi[1](data[i]), derivative(u, phi, ....eps), θ[end-1]`.

Reply:
Maybe it makes sense to have the symbolic optimization condition.
Something like that:
```julia
@parameters t_d, x_d, y_d, z_d
optim_param_condition = [x(t_d) - x_d,
                                             y(t_d) - z_d,
                                             z(t_d)) - y_d]
EstimatiomParametersProblem(optim_param_condition,[t_d,x_d,y_d,z_d], data)
```

Reply:
We shouldn't add another Problem type. It doesn't match the interface.

Reply:
>as @zoemcc already mentioned, if this is an additional general purpose function then it should not depend on param_estim if not and this is a function only for parameter optimization then it should be renamed.

Agreed on this.